starter-code/MyLinkedList.java

```diff
@@ -13,14 +13,39 @@ public int getSize() {
 
 	public boolean remove(int index) {
 		//to-do
+		if (index > size) {
+			return false;
+		}
+		Node<T>node = head;
+		Node<T>afterNode;
+		for (int i = 0; i < index-1; i++) {
+			node = node.getNext();
+		}
+		afterNode = node.getNext();
+		node.setNext(afterNode);
+		return true;
+
 	}
 
 	public T get(int index) {
 		//to-do
+		Node<T> nodeAtIndex = head;
+		for (int i = 0; i <index ; i++) {
+			nodeAtIndex = returnedNode.getNext();
+		}
+		return nodeAtIndex.getData();
```

Comment:
There are a couple of situations where you could get a null pointer exception here. The first is if head is null (i.e. size is zero) and you call `getData()` on a null object. The other is if `index > size`, then at some point in the loop body `nodeAtIndex` would go past the end of the list and be null.